File: persephone/provision-service/test/jvm/com/vmware/blockchain/deployment/service/provision/TestOrchestratorModule.java
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc.  All rights reserved. VMware Confidential
 * *************************************************************************/
package com.vmware.blockchain.deployment.service.provision;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class TestOrchestratorModule {

    /**
     * Provide an {@link OrchestratorProvider} instance.
     *
     * @return
     *   a singleton {@link OrchestratorProvider} instance.
     */
    @Provides
    @Singleton
    static OrchestratorProvider providesOrchestratorProvider() {
        return new TestOrchestratorFactory();
    }
}

File: persephone/provisioning-service/src/test/java/com/vmware/blockchain/services/provision/TestOrchestratorModule.kt
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc. All rights reserved. VMware Confidential
 * **************************************************************************/
package com.vmware.blockchain.services.provision

import com.vmware.blockchain.deployment.orchestration.OrchestratorProvider
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TestOrchestratorModule {

    /**
     * Provide an [OrchestratorProvider] instance.
     *
     * @return
     *   a singleton [OrchestratorProvider] instance.
     */
    @Provides
    @Singleton
    fun providesOrchestratorProvider(validator: com.vmware.blockchain.services.provision.OrchestrationSiteValidator): OrchestratorProvider {
        return com.vmware.blockchain.services.provision.TestOrchestratorFactory(validator)
    }
}
